crypto/async/async.c

```diff
@@ -203,7 +203,10 @@ int ASYNC_start_job(ASYNC_JOB **job, ASYNC_WAIT_CTX *wctx, int *ret,
             }
 
             if (ctx->currjob->status == ASYNC_JOB_PAUSED) {
+                if (*job == NULL)
```

Comment:
It looks like it only gets here if the param passed into the function (*job) is NULL?
Is this a valid scenario?

Reply:
Refer to @mattcaswell's comment: https://github.com/openssl/openssl/pull/14589#discussion_r595890658